src/io_uring.rs

```diff
@@ -1047,6 +1128,18 @@ pub struct io_uring_buf {
     pub resv: u16,
 }
 
+impl Default for ioprio_or_flags_union {
+    #[inline]
+    fn default() -> Self {
+        let mut s = ::core::mem::MaybeUninit::<Self>::uninit();
```

Comment:
`::zeroed()` would be much clearer instead of the ptr write.

Reply:
I believe the original idea for `write_bytes` in other places that do this is that `zeroed()` doesn't guarantee to write to any padding bytes, while `write_bytes` does. However, looking at it now, this doesn't seem like it should matter. This `default()` function still relies on Rust to move the value into the return value, which possibly doesn't guarantee to preserve padding bytes, in which case we could just as well use `zeroed()` instead.

Reply:
It seems like the UCG WG hasn't fully flushed out union semantics yet, but Ralf is leaning towards making it bag-o-bytes. Either way, zeroed can't leave padding bytes that are potentially used by one of the variants in an initialized state since it's a union and you don't know which variant will be used. If there are uninit padding bytes, they'd have to be unused by all variants (which seems like what currently happens: https://github.com/rust-lang/unsafe-code-guidelines/issues/354#issuecomment-1191826685)

---

I wonder if it's just better to return `Self { flags: 0 }`. You could argue that if the union is ever expanded, then it's remaining bytes will be uninitialized (I think) which is bad, but then again it's a union and you have no idea what variant it contains anyway. Also Linux can't change it's API so it seems pretty safe to just say flags: 0 and call it good.

Reply:
I am more a fan of using `::mem::zeroed()` as functionally it does the same thing as `Self { ioprio: 0 }` (note: changed the field name). But it won't break if the union is ever extended. As what sunfishcode said makes sense, changed this to be how all unions do the defaults.

I'm not sure if the padding bytes not being zeroed is an issue anyway. Only the unions that are fields of `io_uring_sqe` implement default. As those are going to be aligned correctly, I don't see why the kernel would be touching those uninit padding bytes. As `zeroed()` will zero the largest variant of the union.